ruby/ql/lib/codeql/ruby/dataflow/FlowSummary.qll

```diff
@@ -108,6 +131,17 @@ abstract class SummarizedCallable extends LibraryCallable {
   predicate clearsContent(int i, DataFlow::Content content) { none() }
 }
 
+/**
+ * A callable with a flow summary, identified by a unique string, where all
+ * calls to a method with the same name are considered relevant.
+ */
+abstract class SimpleSummarizedCallable extends SummarizedCallable {
+  bindingset[this]
+  SimpleSummarizedCallable() { any() }
+
+  final override MethodCall getACall() { result.getMethodName() = this }
```

Comment:
This sounds like it will match _any_ method with the given name. That seems too broad? What happens when we want to add a flow summary for `Hash#assoc`, which behaves entirely differently from `Array#assoc`?

Reply:
That's a good question. @hvitved ?

Reply:
Indeed, it will. However, almost all summaries read from an array element from the input, so in practice they will only apply if data is indeed stored inside a collection of the argument, and if that is the case the summary is probably correct.

Reply:
So for `assoc` this means that, once we also model hashes, we will synthesize just one `assoc` method, but there will be two flow paths through it: one that reads reads array elements and one that reads hashes, and only one of them will apply, depending on how data is stored in the receiver.

Reply:
When we have models as data, will the same behaviour apply? It seems inevitable that we will add summaries for two different libraries which have the same method name, and they might not be as easily distinguishable as the `Hash/Array#assoc` case.

Reply:
For models-as-data we will identify calls using API graphs, so that will be much more restricted.